### DISCUSSION TITLE
Enhance object selection

### DISCUSSION THREAD

**Discussion by PaulR282** [netbox-community/netbox#12695] [Ideas]
The new object selector makes creating new objects e.g. cables take much longer when you have a lot of devices. Patchpanels typically don't have a unique name - so you can't search for a patch panel without the selector. Even in the selector, it takes far more steps to get the right device (Like 10x longer than before). This becomes a big problem when you want to document a lot of cables.
Another problem is #12684. For me, this is still a bug because it does not implement the feature, that an unnamed VC member gets named after the VC with : VC Position. 

What do you think? Anyone else having problems with the new selector?

My thoughts on solutions:
- preselect the location and rack when searching for a device when e.g. creating a cable like it was pre v3.5
- make VCs appear in the search results

**Comment by PaulR282** [comment]
Another FR of this: #12705

**Comment by PaulR282** [comment]
Another idea: Being able to connect to different types (FP, Interface) - also in the same selection.

#13146

**Comment by dywilson73** [comment]
I think the new object selector offers great flexibility. But it definately needs to be refined to make repetitive entry quicker. 
It is fine for adding in one cable, but adding a whole 24 port patch panel became a lot more expensive - time wise.

Id love to see this happen soon simply because we are seeing thing not being documented becouse of the time friction. 

There are actually quite a lot of areas where time improvements could be made. 

**Logical / automatic default filters** : reduces the drop-downs to the most likely subset - exactly what is described above
**configurable defaults** : allows users to preset the defaults that they use the most (such as defaulting cable length to metres) for us as we only ever enter lengths as meters 
**excluding entries from a dropdown** : this saves us scrolling through a list of items that we never use. Such as interface types. There are a lot if interfaces that people may use. But there are often also quite a number that they would never use and so it makes sense to remove them - they can always add them back in if they ever do use them in the future.
